### ISSUE TITLE
Remove Unspecified from user type

### ISSUE THREAD

**Issue by Castronova** [hydroshare/hydroshare#3812]
The user type drop menu on the profile page contains "unspecified". This should be removed since we already have an "other" option. Removing this will help improve our metrics/reporting

![image](https://user-images.githubusercontent.com/4822372/78915533-123f3900-7a5a-11ea-8ca1-e0f986d7ff39.png)

**Comment by horsburgh** [comment]
@Castronova - is this simply removing the item from the list or does it also include updating any users for which the option "Unspecified" has been chosen? Is this something @Maurier could do really quickly?

**Comment by Castronova** [comment]
@horsburgh I think it can be done in two steps. The first would be to remove this option from the list which will take @Maurier 5 minutes to complete. The more difficult step would happen afterwards...removing the Unspecified type from existing user profiles. This might be a combination of notifying users and implementing code that forces them to change their user type. As far as correcting the profile data goes, I suggest that it be broken out into a separate issue.

**Comment by dtarb** [comment]
@Castronova @horsburgh  for what it is worth back in the day (2013-2016  era) when use tracking and profiles were conceived both "Unspecified" and "Other" were deliberate.  See https://docs.google.com/document/d/1VhKhxeuBZb6s5K3mPuQnNx6JSO0JIW5T6pi09zL4AjI/edit

Unspecified is the default when an account is created.  "Other" is different from "Unspecified".  When a user selects Other, there is a required free form text field to specify what Other refers to.

If you remove unspecified, you will need to make a choice as to what to default new accounts to.  If the default is "Other" then you will need to fill in the required text field.  I do not think it is a good idea to default to any of the other options.

Unless you want to add "User Type" to the form where an account is created, I do not think it will be easy to get rid of "Unspecified".  I see three choices, that are essentially policy choices

(A) Add filling in (or selecting) user type to the account creation form.  Then this information will be captured with account creation.  For consistency other required information (currently Country and State are required - to facilitate metrics on geographic use) could be captured here too - adopting the policy that if it is required, we ask for it up front.  Anything else in the profile is optional.
(B) Leave things as they are but implement reduced functionality for users with incomplete profiles.  E.g. creating a resource, joining a group might check for profile completeness and disallow the action with a message that profile information needs to be completed before the action is allowed.
(C) Rework the account creation process so that it has steps (1) sign up, (2) verify email, (3) land on profile page in edit mode and click Save.  Unless you complete (3) and save changes with all required fields filled out the account is not activated.  Any attempts to log in with an inactive account but verified email take you to profile page to complete required information.

So, sorry to rain on the parade, but I fear removing unspecified is not that simple.  If you really want, I could dig out some back in the day workflows for ideas on these designs, but rather than revisiting these, I would suggest a fresh and holistic look at the account creation process and information needed.

**Comment by horsburgh** [comment]
I will defer to @Castronova on what is needed here. Tony - once the policy decisions are made, Mauriel could probably help with the implementation.

**Comment by Castronova** [comment]
@dtarb Thanks for linking the design doc. Currently, most of our users have the "unspecified" user type which is problematic when compiling our metrics for NSF. Given, that we have an "other" option, I propose that we remove "unspecified" since it's not meaningful.  According to the user and group profile document (https://docs.google.com/document/d/1l4zJlGyeq2WONijr-azZvFlcG9uvGKXeHJHDcEtFk3Q/edit) it appears that User Type is a required field when editing and saving profile information, but not when creating an account. Since we give all users an "unspecified" user type upon account creation it's not really acting like a required field. Because of this, HS users that have provided and alternative user type have gone into their profile and updated sometime after they created their account. I don't think this is what we had envisioned back in 2016 when the User and Group Profile design document was created. To make this field more informative, we should not specify a default value and instead require users to choose a user type during account creation. Of course, this would require additional work on the account creation page. 

I'm looping in @jtpollak since he's responsible for writing out NSF reports and I would like to hear his thoughts on this.

**Comment by dtarb** [comment]
@Castronova what you outlined is the option (A) alternative I mentioned above.  It will require entry of user type during account creation.  It is certainly doable.  But the solution does require changing the login process in addition to just removing this user type from profile, so is not as simple as the 5 min suggested above.

**Comment by jtpollak** [comment]
@dtarb @Castronova @horsburgh I would enthusiastically support requiring User Type during the account creation process (i.e. Option A above). In order to deal with a legacy data, could we force users that have an Unspecified User Type to select one as they login? What I am envisioning is a pop-up with a User Type selection form that doesn't allow the user to proceed with doing anything in HydroShare until the user selects a type.

**Comment by Castronova** [comment]
I created #3838 to capture the account creation requirements.

**Comment by Castronova** [comment]
@horsburgh I created ticket #3838 to capture the requirements for modifying the account creation page. This is required before completing this ticket, i.e. removing Unspecified from the user type list.

**Comment by hyi** [comment]
@Castronova I created a PR #3972 to address the ticket #3838 and asked you to code review. In the PR, unspecified is already removed from user profile editing so that users cannot select unspecified any more when editing their profile. So what is left to do in this issue is how to handle existing users with unspecified user type. Do we want to notify those users with a popup asking them to select a valid user type before allowing them to do anything else? Have we made a decision how to handle this case? Let me know what the decision is to handle this before I can start implementation.
